quantus/metrics/base.py

```diff
@@ -1,105 +1,331 @@
+    def __init__(
+            self,
+            abs: bool = False,
+            normalise: bool = False,
+            normalise_func: Optional[Callable] = None,
+            normalise_func_kwargs: Optional[Dict[str, Any]] = None,
+            softmax: bool = False,
+            default_plot_func: Optional[Callable] = None,
+            disable_warnings: bool = False,
+            display_progressbar: bool = False,
+            **kwargs):
         """
         Initialise the Metric base class.
 
         Each of the defined metrics in Quantus, inherits from Metric base class.
 
-        To add a new metric classes to the library, the minimum set of attributes that should be included are:
-
-            args: a arguments (optional)
-            kwargs: a dictionary of key, value pairs (optional)
-            abs: a bool stating if absolute operation should be taken on the attributions
-            normalise: a bool stating if the attributions should be normalised
-            normalise_func: Callable that make a normalising transformation of the attributions
-            default_plot_func: Callable that plots the metrics result
-            last_results: a list containing the resulting scores of the last metric instance call
-            all_results: a list containing the resulting scores of all the calls made on the metric instance
+        Parameters
+        ----------
+        abs (boolean): Indicates whether absolute operation is applied on the attribution, default=False.
+        normalise (boolean): Indicates whether normalise operation is applied on the attribution, default=True.
+        normalise_func (callable): Attribution normalisation function applied in case normalise=True.
+            If normalise_func=None, the default value is used, default=normalise_by_negative.
+        normalise_func_kwargs (dict): Keyword arguments to be passed to normalise_func on call, default={}.
+        softmax (boolean): Indicates wheter to use softmax probabilities or logits in model prediction, default=False.
+        default_plot_func (callable): Callable that plots the metrics result.
+        disable_warnings (boolean): Indicates whether the warnings are printed, default=False.
+        display_progressbar (boolean): Indicates whether a tqdm-progress-bar is printed, default=False.
 
         """
-        self.args = args
-        self.kwargs = kwargs
-        self.abs = self.kwargs.get("abs", False)
-        self.normalise = self.kwargs.get("normalise", False)
-        self.normalise_func = self.kwargs.get("normalise_func", normalise_by_negative)
-        self.default_plot_func = Callable
-        self.disable_warnings = self.kwargs.get("disable_warnings", False)
-        self.display_progressbar = self.kwargs.get("display_progressbar", False)
+        self.abs = abs
+        self.normalise = normalise
+        self.softmax = softmax
+
+        if normalise_func is None:
+            self.normalise_func = normalise_by_negative
```

Comment:
Why not add to signature? much clearer that `normalise_by_negative` is the default value then